compiler/flops.cc

```diff
@@ -32,22 +32,66 @@ int64_t OutputSize(const Node& node) {
 }
 
 int64_t CalculateFlopsOfGemm(const Node& node) {
-    int64_t flops = node.output(0)->type().NumElements();
+    int64_t const out_size = node.output(0)->type().NumElements();
+    int64_t flops = out_size;
     flops *= node.input(0)->type().dims()[node.trans_a() ? 0 : 1];
+    if (node.alpha() != 1.0) {
+        flops += out_size;
+    }
+    if (node.beta() != 0.0) {
+        flops += out_size * node.input(2)->type().dims()[node.trans_a() ? 0 : 1];
+        if (node.beta() != 1.0) {
```

Comment:
I guess we don't need this extra cost for `0 < beta < 1` since it could be done by FMA?

Reply:
So can I expect it to be fused in the future?
(Current implementation isn't fused:
https://github.com/pfnet-research/chainer-compiler/blob/db3e2ec3aedf8654b7f730c11b7ce8c934322a14/runtime/ops/math.cc#L162

Reply:
Good point. Yeah, I feel the current implementation is just inadequate. I think we should just call sgemm function in OpenBLAS or cuBLAS, which would have the sane implementation, I believe. I think it'd be better to show flops of sane implementation since this number will be used to estimate theoritical limit latency.